Fix hardcoded dashboard expiration date format

masteradhoc opened the pull request:
Fixes #2890

### Changes Proposed in this Pull Request
* We should change the hardcoded dateformat for the Job Dashboard Shortcode. Currently it displays the date format hardcoded in the format 'M d, Y'. Now it will respect the settings set in WordPress.

### Testing Instructions
* open /job-dashboard/ page. without the fix it will be displayed like this
<img width="1610" height="365" alt="image" src="https://github.com/user-attachments/assets/d19f4eea-b87c-4064-8c9d-73fb3e506969" />

After the fix it will be displayed according to your settings, in my example "j. F Y".
<img width="1616" height="273" alt="image" src="https://github.com/user-attachments/assets/5cbdf6e0-f68a-43c7-93a3-1d0ac98f9ca6" />

<!-- Add changelog entries meant for end-users. Leave empty to skip changelog. Delete section to use PR title. -->
### Release Notes
* fix hardcoded dashboard expiration date format

### New or Updated Hooks and Templates
<!-- Add notes for developers on hook/template changes. Add the following only if there are new/updated actions or filters. Please provide a brief description of what they do and any arguments they may take. Be sure to also add the "Hooks" label to this PR. -->
* -

<!-- Add the following only if there is any code that is being deprecated. Please list the replacement function or hook that should be called instead, if applicable. Be sure to also add the "Deprecation" label to this PR. -->
### Deprecated Code
* -

### Screenshot / Video
-

masteradhoc committed:
fix hardcoded dashboard expiration date format #2890

azaozz reviewed (changes requested):
I'm pretty sure it is (almost) impossible to not have the `date_format` option set in WP, but just in case that happens somewhere this `get_option()` call will result in a trip to the DB (slow) and will return `false` which seems undesirable.

How about checking the option value before using it? Perhaps add something like:

`$date_format = get_option( 'date_format' ) ?: 'M d, Y';`

at the top then use `$date_format` in the HTML.

azaozz committed:
Merge remote-tracking branch 'upstream/trunk' into pr/2891

masteradhoc committed:
implement suggested changes, fix docblock

masteradhoc commented:
@azaozz definately a valid point! adjusted the code as per your suggestion and fixed the docblock as the title was also wrong there. hope thats fine inside the same PR.

azaozz reviewed (approved):
Looks good, thanks!
